lametro/views.py

```diff
@@ -130,15 +130,19 @@ def post(self, request, *args, **kwargs):
 
     def get_context_data(self, **kwargs):
         context = super(EventDetailView, self).get_context_data(**kwargs)
-        # Create URL for packet download.
+        
+        # Get the event with prefetched media_urls in proper order. 
         event = context['event']
+        event = LAMetroEvent.objects.with_media().filter(ocd_id=event.ocd_id).first()
+        context['event'] = event
```

Comment:
this looks like a good case for overriding [the `get_object` method](https://ccbv.co.uk/projects/Django/2.1/django.views.generic.detail/DetailView/#get_object)!

Reply:
GOOD IDEA